pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -397,8 +397,9 @@ def reduce(self, tensor, group: Optional[Any] = None, reduce_op: Union[ReduceOp,
             tensor = sync_ddp_if_available(tensor, group, reduce_op=reduce_op)
         return tensor
 
-    def training_step(self, *args, **kwargs) -> Optional[Any]:
-        return self.model(*args, **kwargs)
+    def training_step(self, step_kwargs: Dict[str, Union[Any, int]]) -> STEP_OUTPUT:
+        with self.precision_plugin.train_step_context():
+            return self.model(*step_kwargs.values())
```

Comment:
Depending on the order they were assigned to the dict, this may not be the same. Also in general, there may be keyword only arguments